files/scripts/syncd.sh

```diff
@@ -46,6 +46,10 @@ function startplatform() {
 function waitplatform() {
     
     if [[ x"$sonic_asic_platform" == x"mellanox" ]]; then
+        BOOT_TYPE=`getBootType`
+        if [[ x"$BOOT_TYPE" == x"fast" ]]; then
+            return
+        fi
         debug "Starting pmon service..."
         /bin/systemctl start pmon
         debug "Started pmon service"
```

Comment:
This was added by https://github.com/Azure/sonic-buildimage/pull/3505 to fix some issues related to pmon/SDK. Won't that issue happen again with your change?

Reply:
This change can be made generic to all platforms. CC @yxieca.

Also, if this is not made generic, then the comments on `files/build_templates/pmon.timer` should be modified to say:
`# This delay is for fast-reboot performance on MLNX platform`

Reply:
Also, don't you want to do this change for warm-reboot as well? I am curious why only for fast-reboot?

Reply:
It is a generic change, this exception is only for MLNX platform, so I added a condition for it in order to keep the delay and not start PMON with SYNCD start flow.
Regarding warm-boot, I totally agree we can do it for both types, fixed it.